operator/pkg/applier/kubectl.go

```diff
@@ -141,13 +152,24 @@ func (c *Kubectl) Apply(targetNamespace string, yamlDoc []byte, dryRun bool, wai
 		}...)
 	}
 
-	args = append(args, []string{
-		"-f",
-		"-",
-	}...)
+	yamlPath := filepath.Join(tmp, "doc.yaml")
+	if err := ioutil.WriteFile(yamlPath, yamlDoc, 0644); err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to write %s", yamlPath)
+	}
+
+	kustomizationPath := filepath.Join(tmp, "kustomization.yaml")
+	kustomizationYaml := fmt.Sprintf(`
+resources:
+- doc.yaml
+
+commonAnnotations:
+  kots.io/app-slug: %q
```

Comment:
would it be better to add this annotation on the kotsadm side of things?

Reply:
Is `kustomize` available in kotsadm?